tools/topology/topology1/sof-tgl-max98357a-rt5682.m4

```diff
@@ -282,6 +282,12 @@ ifelse(
 		SSP_CLOCK(fsync, 48000, codec_slave),
 		SSP_TDM(2, 16, 3, 3),
 		SSP_CONFIG_DATA(SSP, SPK_SSP_INDEX, 16)))',
+	CODEC, `MAX98360A', `
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
+		SSP_CLOCK(bclk, 2304000, codec_slave),
```

Comment:
does this work? this would mean 48 bits per sample, and is not an even division of 19.2 (8.333).

Why not use 2400000 and 50 bits per sample. as done in all other cases?

Reply:
It's I2S 48fs so each channel is 24bit. According to the datasheet, the I2S 24-bit sample resolution only supports 48fs setting. I will check with Maxim to see if 50fs is acceptable.

Reply:
in the past 50.fs was a problem with Maxim devices, so we used 1536000 or 3072000, with the 24.576 MHz PLL selected as source instead of the 38.4 MHz oscillator.

Reply:
I checked the datasheet, the i2s supports 32/48/64 fs in I2S mode. And we use 48fs (2.304MHz) on JSL platform.

Reply:
@plbossart I'm assuming the problem here is that we don't want to use the m/n in this case ?

Reply:
yes, I don't understand how we support 48 bits per slot. It's an oversight, we should NEVER have let this happen. The M/N solution will have a material impact on the duty cycle and that can have impacts on audio quality. 'validated by odm' is not good enough for me, unless we get audio quality measurements we have to assume it's broken.

The only occurrences of this are the following:
tools/topology/topology1/platform/intel/cml.m4:define(`SSP1_BCLK', `2304000')
tools/topology/topology1/platform/intel/icl.m4:    define(`SSP1_BCLK', 2304000),
tools/topology/topology1/platform/intel/jsl-dedede.m4:                        SSP_CLOCK(bclk, 2304000, codec_slave),
tools/topology/topology1/platform/intel/whl.m4:    define(`SSP1_BCLK', 2304000),

Reply:
If we want to avoid m/n counter then we can use 32/64fs which BCLK frequency is 1.536/3.072MHz. In this case, the cardinal clock (24.576MHz) will be used. I will upload a 64fs version.